docs/releases/upcoming/1738.bugfix.rst

```diff
@@ -0,0 +1 @@
+Convert ``traitsui.instance_choice.InstanceChoiceItem`` into an instance of ``traits.api.ABCHasStrictTraits`` to avoid users from instantiating the object directly.
```

Comment:
I consider this pull request deserves a news entry, because some users might be interested on instantiating this object directly.

Reply:
@rahulporuri what do you think about adding this news file?